conan/api/subapi/cache.py

```diff
@@ -101,6 +109,72 @@ def clean(self, package_list, source=True, build=True, download=True, temp=True)
+        app = ConanApp(cache_folder)
+        out = ConanOutput()
+        name = os.path.basename(tgz_path)
+        with open(tgz_path, "wb") as tgz_handle:
+            tgz = gzopen_without_timestamps(name, mode="w", fileobj=tgz_handle)
+            for ref, ref_bundle in package_list.refs().items():
+                ref_layout = app.cache.recipe_layout(ref)
+                recipe_folder = os.path.relpath(ref_layout.base_folder, cache_folder)
+                ref_bundle["recipe_folder"] = recipe_folder
+                out.info(f"Saving {ref}: {recipe_folder}")
+                tgz.add(os.path.join(cache_folder, recipe_folder), recipe_folder, recursive=True)
+                for pref, pref_bundle in package_list.prefs(ref, ref_bundle).items():
+                    pref_layout = app.cache.pkg_layout(pref)
+                    pkg_folder = pref_layout.package()
+                    folder = os.path.relpath(pkg_folder, cache_folder)
+                    pref_bundle["package_folder"] = folder
+                    out.info(f"Saving {pref}: {folder}")
+                    tgz.add(os.path.join(cache_folder, folder), folder, recursive=True)
+                    if os.path.exists(pref_layout.metadata()):
+                        metadata_folder = os.path.relpath(pref_layout.metadata(), cache_folder)
+                        pref_bundle["metadata_folder"] = metadata_folder
+                        out.info(f"Saving {pref} metadata: {folder}")
+                        tgz.add(os.path.join(cache_folder, metadata_folder), metadata_folder,
+                                recursive=True)
+            serialized = json.dumps(package_list.serialize(), indent=2)
+            info = tarfile.TarInfo(name="pkglist.json")
+            data = serialized.encode('utf-8')
+            info.size = len(data)
+            tgz.addfile(tarinfo=info, fileobj=BytesIO(data))
+            tgz.close()
+
+    def restore(self, path):
+        with open(path, mode='rb') as file_handler:
+            the_tar = tarfile.open(fileobj=file_handler)
+            fileobj = the_tar.extractfile("pkglist.json")
+            pkglist = fileobj.read()
+            the_tar.extractall(path=self.conan_api.cache_folder)
+            the_tar.close()
+
+        out = ConanOutput()
+        package_list = PackagesList.deserialize(json.loads(pkglist))
+        cache = ClientCache(self.conan_api.cache_folder)
+        for ref, ref_bundle in package_list.refs().items():
+            ref.timestamp = revision_timestamp_now()
+            recipe_layout = cache.get_or_create_ref_layout(ref)
+            recipe_folder = ref_bundle["recipe_folder"]
+            rel_path = os.path.relpath(recipe_layout.base_folder, cache.cache_folder)
+            assert rel_path == recipe_folder, f"{rel_path}!={recipe_folder}"
+            out.info(f"Put: {ref} in {recipe_folder}")
+            for pref, pref_bundle in package_list.prefs(ref, ref_bundle).items():
+                pref.timestamp = revision_timestamp_now()
```

Comment:
If I do `conan cache save zlib/1.3:*#*`, does this mean that it forgets which one is the "latest"?

Reply:
That is a very good point. I think we need to make sure  we are keeping the order, to make sure they are restored in the same order. I'll do it.

Reply:
Being based in package-lists, it keeps revisions in order, restoring them in order. I have added a test that validates the order for recipe-revision, which would be the most common use case, package-revisions should be way less common.